### PR TITLE
fix(wdl-grammar): false positives in preamble_comment

### LINKED ISSUE #50: Consolidate preamble comment warnings
See an example of a "bad" preamble here: https://github.com/aws-samples/amazon-omics-tutorials/blob/main/example-workflows/gatk-best-practices/workflows/germline-fastqs-to-vcf/sub-workflows/fastq-to-ubam.wdl

I'm not sure who owns this code (whether it's the Broad or Amazon), but they are using double-pound sign comments the way we use them, but _after_ the version declaration. This causes a LintWarning to be thrown for every line of "the preamble" of that file. We should update to detect a continuous preamble _in the wrong location_ and warn once on the whole span.

### PR DESCRIPTION
**Pull request by a-frantz**
This pull request ~adds a new~ modifies a rule in `wdl`.

- **Rule Name**: `preamble_comment`
- **Rule Kind**: Lint warning
- **Rule Code**: `v1::W010`
- **Packages**: `wdl-grammar`

_Describe the rules you have implemented and link to any relevant issues._

Closes #50 . This will only throw one lint warning for continuous blocks of comments. It also permits triple+ pound sign comments.

Before submitting this PR, please make sure:

- [x] You have added a few sentences describing the PR here.
- [x] You have added yourself or the appropriate individual as the assignee.
- [x] You have added at least one relevant code reviewer to the PR.
- [x] Your code builds clean without any errors or warnings.
- [ ] You have added an entry to the relevant `CHANGELOG.md` (see
      ["keep a changelog"] for more information).
- [x] Your commit messages follow the [conventional commit] style.

Rule specific checks:

- [ ] You have added the rule as an entry within the the package-specific rule
      tables (`wdl-ast/src/v1.rs` for AST-based rules and 
      `wdl-grammar/src/v1.rs` for parse tree-based rules).
- [ ] You have added the rule as an entry within the the global rule
      table at `RULES.md`.
- [ ] You have added the rule to the appropriate `fn rules()`.
    - Validation rules added to `wdl-ast` should be added to `fn rules()` within 
      `wdl-ast/src/v1/validation.rs`.
    - Lint rules added to `wdl-ast` should be added to `fn rules()` within `wdl-ast/src/v1/lint.rs`.
    - Validation rules added to `wdl-grammar` should be added to `fn rules()` within 
      `wdl-grammar/src/v1/validation.rs`.
    - Lint rules added to `wdl-grammar` should be added to `fn rules()` within 
      `wdl-grammar/src/v1/lint.rs`.
- [x] You have added a test that covers every possible setting for the rule 
      within the file where the rule is implemented.
- [ ] You have run `wdl-gauntlet --refresh` to ensure that there are no unintended
      changes to the baseline configuration file (`Gauntlet.toml`).
- [x] You have run `wdl-gauntlet --refresh --arena` to ensure that all of the rules
      added/removed are now reflected in the baseline configuration file
      (`Arena.toml`).

[conventional commit]: https://www.conventionalcommits.org/en/v1.0.0/#summary
["keep a changelog"]: https://keepachangelog.com/en/1.0.0/

### CODE REVIEW

**Comment by a-frantz** [comment]
On Slack we discussed some other comment cases we wanted to throw warnings on. Should this rule be expanded to cover those cases? Or should those additional cases go into another rule? I think as a general rule we want to prefer multiple targeted rules over fewer broad rules.

We've already split this rule (preamble_comment) from the document_preamble rule. But maybe we should only have one rule for the preamble??? I'm not sure how we want to divvy up/classify these warnings.

File: wdl-grammar/src/v1/lint/preamble_comment.rs [hunk 1]
```diff
@@ -135,7 +166,30 @@ version 1.0
...
+        );
+        Ok(())
+    }
+
+    #[test]
+    fn it_catches_badly_formatted_preamble_comment_block() -> Result<(), Box<dyn std::error::Error>>
+    {
+        let tree = Parser::parse(
+            Rule::document,
+            r#"# a comment
```
Comment (adthrasher) [inline wdl-grammar/src/v1/lint/preamble_comment.rs:179]: So we're explicitly disallowing single pound comments in the preamble?

Comment (a-frantz) [reply wdl-grammar/src/v1/lint/preamble_comment.rs:179]: Yes. That's not new in this PR. I thought we had a discussion about it back when this rule was first implemented.

Comment (adthrasher) [reply wdl-grammar/src/v1/lint/preamble_comment.rs:179]: In the `document_preamble` lint check, I only look for `COMMENT` before the version. For some reason, I thought this rule (`preamble_comment`) was only about disallowing the double-pound syntax outside of the preamble.

Comment (adthrasher) [reply wdl-grammar/src/v1/lint/preamble_comment.rs:179]: I don't think we should change it if we've discussed previously. I was just confirming that this was our intention here.

**Comment by adthrasher** [comment]
Did we decide on whether no spacing after a `#` is allowable, e.g. `##this is a comment`? I'm guessing if so, that should be a separate rule.

File: wdl-grammar/src/v1/lint/preamble_comment.rs (hunk 1 above)
Comment (a-frantz) [reply wdl-grammar/src/v1/lint/preamble_comment.rs:179]: The fact you had to ask to clarify whether this was an intended case or not is more evidence that we need to write a `body` (as defined in #12 ) for each rule. The current situation is that you need to code dive to discern what exactly a rule covers. We should have it written in plain text somewhere.

**Review by adthrasher — APPROVED** [review]

**Pull request merged**